File: Easy/Java/566. Reshape the Matrix.java
class Solution {
    public int[][] matrixReshape(int[][] mat, int r, int c) {
        int m = mat.length;
        int n = mat[0].length;
        if(r * c != m * n)
            return mat;
        
        int[][] rs = new int[r][c];
        for(int i = 0; i < r * c; i++){
            rs[i / c][i % c] = mat[i / n][i % n];
        }

        return rs;
    }
}

File: Easy/Python/566. Reshape the Matrix.py
class Solution(object):
    def matrixReshape(self, mat, r, c):
        m = len(mat)
        n = len(mat[0])
        if r * c != m * n:
            return mat
        
        rs = [[0] * c for _ in range(r)]
        for i in range(r * c):
            rs[i / c][i % c] = mat[i / n][i % n]

        return rs
